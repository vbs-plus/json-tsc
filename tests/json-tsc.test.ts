import { describe, expect, test } from 'vitest'
import { JSON_TSC } from 'json-tsc'

const JSON_TEST = {
  BigIntSupported: 99581589,
  date: '20180322',
  message: 'Success !',
  status: 200,
  city: '北京',
  count: 632,
  data: {
    shidu: '34%',
    pm25: 73,
    pm10: 91,
    quality: '良',
    wendu: '5',
    ganmao: '极少数敏感人群应减少户外活动',
    yesterday: {
      date: '21日星期三',
      sunrise: '06:19',
      high: '高温 11.0℃',
      low: '低温 1.0℃',
      sunset: '18:26',
      aqi: 85,
      fx: '南风',
      fl: '<3级',
      type: '多云',
      notice: '阴晴之间，谨防紫外线侵扰',
    },
    forecast: [
      {
        date: '22日星期四',
        sunrise: '06:17',
        high: '高温 17.0℃',
        low: '低温 1.0℃',
        sunset: '18:27',
        aqi: 98,
        fx: '西南风',
        fl: '<3级',
        type: '晴',
        notice: '愿你拥有比阳光明媚的心情',
      },
    ],
  },
}

describe('json-tsc transform', () => {
  test('show work with default config', () => {
    const json_tsc = new JSON_TSC({
      prependWithO: true,
      sortAlphabetically: false,
      prependExport: true,
      useArrayGeneric: false,
      optionalFields: false,
      prefix: '',
    })

    const result = json_tsc.transform(JSON_TEST)

    expect(result).toMatchSnapshot()
  })

  test('show work with custom config', () => {
    const json_tsc = new JSON_TSC({
      prependWithO: true,
      sortAlphabetically: true,
      prependExport: false,
      useArrayGeneric: true,
      optionalFields: true,
      prefix: 'TEST',
    })
    const result = json_tsc.transform(JSON_TEST)

    expect(result).toMatchSnapshot()
  })
})
