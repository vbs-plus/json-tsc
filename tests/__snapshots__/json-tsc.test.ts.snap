// Vitest Snapshot v1

exports[`json-tsc transform > show work with custom config 1`] = `
"type JSON_TSC = OTESTRootObject

interface OTESTRootObject {
  BigIntSupported?: number;
  city?: string;
  count?: number;
  data?: OTESTData;
  date?: string;
  message?: string;
  status?: number;
}

interface OTESTData {
  forecast?: Array<OTESTForecast>;
  ganmao?: string;
  pm10?: number;
  pm25?: number;
  quality?: string;
  shidu?: string;
  wendu?: string;
  yesterday?: OTESTYesterday;
}

interface OTESTYesterday {
  aqi?: number;
  date?: string;
  fl?: string;
  fx?: string;
  high?: string;
  low?: string;
  notice?: string;
  sunrise?: string;
  sunset?: string;
  type?: string;
}

interface OTESTForecast {
  aqi?: number;
  date?: string;
  fl?: string;
  fx?: string;
  high?: string;
  low?: string;
  notice?: string;
  sunrise?: string;
  sunset?: string;
  type?: string;
}
"
`;

exports[`json-tsc transform > show work with default config 1`] = `
"type JSON_TSC = ORootObject

export interface ORootObject {
  BigIntSupported: number;
  date: string;
  message: string;
  status: number;
  city: string;
  count: number;
  data: OData;
}

export interface OData {
  shidu: string;
  pm25: number;
  pm10: number;
  quality: string;
  wendu: string;
  ganmao: string;
  yesterday: OYesterday;
  forecast: OForecast[];
}

export interface OYesterday {
  date: string;
  sunrise: string;
  high: string;
  low: string;
  sunset: string;
  aqi: number;
  fx: string;
  fl: string;
  type: string;
  notice: string;
}

export interface OForecast {
  date: string;
  sunrise: string;
  high: string;
  low: string;
  sunset: string;
  aqi: number;
  fx: string;
  fl: string;
  type: string;
  notice: string;
}
"
`;
